src/openforms/appointments/renderer.py

```diff
@@ -26,5 +30,24 @@ def get_children(self) -> Iterator[ComponentNode]:
                 renderer=self,
                 configuration_path=f"{index}",
             )
-            yield child_node
             yield from child_node
+
+    def __str__(self) -> SafeString:
+        # Like a Django Form renderer, render to the default representation
```

Comment:
I'm not sure I agree - there is the `render_report` management command which loops over all nodes: `for node in renderer:` and then outputs the submission data/information to the CLI (that's the CLI render mode). I would expect to also see the appointment information there.

This implementation now assumes that you'll always be rendering a Django HTML template, but that's not the case. The render nodes implementation abstracts this - output HTML when HTML is requested, output plain text otherwise (which can be for the CLI output mode or the confirmation email...).

I think that `yield (from) AppointmentNode(...)` in the main submission renderer is the better approach here, and then `AppointmentNode` and `AppointmentRenderer` (which is a wart for the e-mails) can share some functionality.